src/front/app/components/Challenges/ChallengeList.js

```diff
@@ -1,7 +1,47 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import Challenge from './Challenge';
+import TopChallenge from './TopChallenge';
 
-export default () => (
-    <div>
-        ChallengeList
-    </div>
-);
+let ChallengeList = ({ challenges, dispatch }) => {
+    if (!challenges.length) {
+        dispatch(getInitialChallenges());
+    }
+    const first = challenges[0];
+    const others = challenges.slice(1);
+
+    return (
+        <ul className="challenges__list">
+            <TopChallenge { ...first } />
+            { others.map((challenge, index) => (
+                    <Challenge key={ index } { ...challenge } />
+            ))}
+        </ul>
+    );
+};
+
+ChallengeList.propTypes = {
+    challenges: React.PropTypes.array,
+    dispatch: React.PropTypes.func
+};
+
+const mapStateToProps = ({ challenges }) => ({ challenges });
+
+function getInitialChallenges() {
+    return function fetchChallenges(dispatch) {
+        fetch('/api/challenges')
+            .then(response => response.json())
+            .then((challenges) => {
+                dispatch({
+                    type: 'GET_INITIAL_CHALLENGES',
+                    challenges
+                });
+            });
+    };
+}
+
+ChallengeList = connect(
+    mapStateToProps
+)(ChallengeList);
+
+export default ChallengeList;
```

Comment:
Consider exporting the result of connect call directly and declare the component as a constant instead of the re-assigning to the variable